trino-lb/src/http_server/admin/mod.rs

```diff
@@ -0,0 +1,170 @@
+    NoAdminAuthenticationMethodDefined,
+
+    #[snafu(display("Invalid admin credentials"))]
+    InvalidAdminCredentials,
+
+    #[snafu(display("Unknown Trino cluster {cluster:?}"))]
+    UnknownCluster { cluster: TrinoClusterName },
+
+    #[snafu(display("Failed to set cluster state for cluster {cluster:?} in persistence"))]
+    SetClusterStateInPersistence {
+        source: trino_lb_persistence::Error,
+        cluster: TrinoClusterName,
+    },
+
+    #[snafu(display("Failed to get all cluster states"))]
+    GetAllClusterStates {
+        source: cluster_group_manager::Error,
+    },
+}
+
+impl IntoResponse for Error {
+    fn into_response(self) -> Response {
+        warn!(error = ?self, "Error while processing admin request");
+        let status_code = match self {
+            Error::NoAdminAuthenticationMethodDefined => StatusCode::UNAUTHORIZED,
+            Error::InvalidAdminCredentials => StatusCode::UNAUTHORIZED,
+            Error::UnknownCluster { .. } => StatusCode::NOT_FOUND,
+            Error::SetClusterStateInPersistence { .. } => StatusCode::INTERNAL_SERVER_ERROR,
+            Error::GetAllClusterStates { .. } => StatusCode::INTERNAL_SERVER_ERROR,
+        };
+        (status_code, format!("{self}")).into_response()
+    }
+}
+
+/// (Re)-Activates a Trino Cluster, so that it receives new queries.
+///
+/// This is useful for maintenance actions (in combination with deactivation).
+#[instrument(name = "POST /admin/activate-cluster/{cluster_name}", skip(state))]
+pub async fn post_activate_cluster(
+    TypedHeader(Authorization(basic_auth)): TypedHeader<Authorization<Basic>>,
+    State(state): State<Arc<AppState>>,
+    Path(cluster_name): Path<TrinoClusterName>,
+) -> Result<Json<ClusterActivationResponse>, Error> {
+    state
+        .metrics
+        .http_counter
+        .add(1, &[KeyValue::new("resource", "post_activate_cluster")]);
+
+    set_cluster_activation(state, basic_auth, &cluster_name, true).await
+}
+
+/// Deactivate a Trino Cluster, so that it doesn't receive any new queries.
+///
+/// This is useful for maintenance actions (in combination with activation).
+#[instrument(name = "POST /admin/deactivate-cluster/{cluster_name}", skip(state))]
+pub async fn post_deactivate_cluster(
+    TypedHeader(Authorization(basic_auth)): TypedHeader<Authorization<Basic>>,
+    State(state): State<Arc<AppState>>,
+    Path(cluster_name): Path<TrinoClusterName>,
+) -> Result<Json<ClusterActivationResponse>, Error> {
+    state
+        .metrics
+        .http_counter
+        .add(1, &[KeyValue::new("resource", "post_deactivate_cluster")]);
+
+    set_cluster_activation(state, basic_auth, &cluster_name, false).await
+}
+
+/// Get the status of the Trino clusters
+#[instrument(name = "GET /admin/cluster-status", skip(state))]
+pub async fn get_cluster_status(
+    State(state): State<Arc<AppState>>,
+) -> Result<Json<BTreeMap<TrinoClusterName, ClusterStats>>, Error> {
```

Comment:
question: Does it make sense to also support retrieving the status of a _single_ cluster via `GET /admin/cluster-status/{cluster_name}`?

Reply:
Sure, if you feel like it.